### PR TITLE
Add Stadia gamepad mapping

### PR DESCRIPTION
**Pull request by ethanlook**
Stadia controllers have a mapping that is different than the default. Controller device names are unique of the form `StadiaXXXX-XXXX` so using the model name to lookup the mapping will fail. Instead, expose a simple helper to get the mapping which checks for the prefix.

**Question:** The mapping for Stadia controllers is the same as the "Xbox Wireless Controller" excluding the "record" button. Should this change reuse the mapping or duplicate it?

**Tested:** Ran from source on a Raspberry Pi 4 with the Stadia gamepad connected by bluetooth. Verified in the Viam app in the `input_controller` control panel.

### CODE REVIEW

File: components/input/gamepad/gamepad_mappings_linux.go [hunk 1]
```diff
@@ -4,14 +4,18 @@
...
 
 	"go.viam.com/rdk/components/input"
 )
 
-// GamepadMappings contains all the axes/button translations for each model.
+const stadiaPrefix = "Stadia"
+
+// gamepadMappings contains all the axes/button translations for each model.
 // use evtest on linux figure out what maps to what.
-var GamepadMappings = map[string]Mapping{
```
Comment (Otterverse) [inline components/input/gamepad/gamepad_mappings_linux.go:14]: Just an FYI, I originally made this public as I figured someone could inject their own mappings in an init() or early main() code back when we were building custom servers for everything. I don't think that paradigm holds today though, so totally fine with this change.

**Review by Otterverse — APPROVED** [review]
LGTM

File: components/input/gamepad/gamepad_mappings_linux.go (hunk 1 above)
Comment (ethanlook) [reply components/input/gamepad/gamepad_mappings_linux.go:14]: Sounds good, thanks for the info.

**Comment by ethanlook** [comment]
@Otterverse do you mind approving the above workflow? It also looks like one of the checks are failing because I'm not a member of the organization. Thank you!

File: components/input/gamepad/gamepad_mappings_linux.go [hunk 2]
```diff
@@ -241,4 +245,40 @@ var GamepadMappings = map[string]Mapping{
...
+			315: input.ButtonStart,
+			316: input.ButtonMenu,
+			317: input.ButtonLThumb,
+			318: input.ButtonRThumb,
+		},
+	},
+}
+
+// GetGamepadMapping returns the mapping for a given model.
+func GetGamepadMapping(model string) (Mapping, bool) {
```
Comment (edaniels) [inline components/input/gamepad/gamepad_mappings_linux.go:277]: this is repetitive naming. can you do `MappingForModel`?

**Review by edaniels — CHANGES_REQUESTED** [review]

File: components/input/gamepad/gamepad_mappings_linux.go (hunk 2 above)
Comment (ethanlook) [reply components/input/gamepad/gamepad_mappings_linux.go]: Done.

**Review by edaniels — APPROVED** [review]

**Comment by Otterverse** [comment]
Thanks for the contribution, @ethanlook !

**Pull request merged**